### PR TITLE
Rename target to resource

### PR DESCRIPTION
**Pull request by fdymylja**
solves #238

### CODE REVIEW

File: cmd/iovnscli/main.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ func main() {
 	}
 
 	// Add --chain-id to persistent flags and mark it required
-	rootCmd.PersistentFlags().String(flags.FlagChainID, "", "Chain ID of tendermint node")
+	rootCmd.PersistentFlags().String(flags.FlagChainID, "", "Chain URI of tendermint node")
```
Comment (davepuchyr) [inline cmd/iovnscli/main.go:62]: ID, not URI, is correct in this case.

File: x/configuration/client/cli/tx.go [hunk 2]
```diff
@@ -138,14 +138,14 @@ func getCmdUpdateConfig(cdc *codec.Codec) *cobra.Command {
...
-				config.ValidBlockchainID = validBlockchainID
+				config.ValidResourceURI = validBlockchainID
 			}
 			validBlockchainAddress, err := cmd.Flags().GetString("valid-blockchain-address")
 			if err != nil {
 				return err
 			}
 			if validBlockchainAddress != defaultRegex {
-				config.ValidBlockchainAddress = validBlockchainAddress
+				config.ValidResourceContent = validBlockchainAddress
```
Comment (davepuchyr) [inline x/configuration/client/cli/tx.go:148]: `ValidResourceContent` should be `ValidResource`.  (The resource's content could be jpeg data, a calendar invite, etc.)

File: x/configuration/client/cli/tx.go [hunk 3]
```diff
@@ -189,12 +189,12 @@ func getCmdUpdateConfig(cdc *codec.Codec) *cobra.Command {
...
 			}
-			blockchainTargetMax, err := cmd.Flags().GetUint32("blockchain-target-max")
+			resourceMax, err := cmd.Flags().GetUint32("resource-max")
 			if err != nil {
 				return err
 			}
-			if blockchainTargetMax != defaultNumber {
-				config.BlockchainTargetMax = blockchainTargetMax
+			if resourceMax != defaultNumber {
+				config.BlockchainResourcesMax = resourceMax
```
Comment (davepuchyr) [inline x/configuration/client/cli/tx.go:197]: `BlockchainResourcesMax` should be `ResourcesMax`.

File: x/domain/account_handlers_test.go [hunk 4]
```diff
@@ -1199,17 +1193,16 @@ func Test_OpenDomain_handleMsgRegisterAccount(t *testing.T) {
 
 func Test_Common_handleMsgRegisterAccount(t *testing.T) {
 	testCases := map[string]keeper.SubTest{
-		"fail invalid blockchain targets address": {
+		"fail invalid blockchain resources address": {
```
Comment (davepuchyr) [inline x/domain/account_handlers_test.go:1196]: "fail invalid resource"

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 # Changelog
 
 ## HEAD
-
+- BREAKING: rename targets to resource
```
Comment (orkunkl) [inline CHANGELOG.md:4]: Move this below under ## Breaking changes section

File: x/domain/account_handlers_test.go [hunk 6]
```diff
@@ -1225,31 +1218,29 @@ func Test_Common_handleMsgRegisterAccount(t *testing.T) {
...
-				if !errors.Is(err, types.ErrInvalidBlockchainTarget) {
-					t.Fatalf("handleMsgRegisterAccount() expected error: %s, got: %s", types.ErrInvalidBlockchainTarget, err)
+				if !errors.Is(err, types.ErrInvalidResource) {
+					t.Fatalf("handleMsgRegisterAccount() expected error: %s, got: %s", types.ErrInvalidResource, err)
 				}
 			},
 			AfterTest: nil,
 		},
-		"fail invalid blockchain targets id": {
+		"fail invalid blockchain resources id": {
```
Comment (davepuchyr) [inline x/domain/account_handlers_test.go:1235]: "fail invalid uri"

File: x/domain/account_handlers_test.go [hunk 7]
```diff
@@ -1321,26 +1312,26 @@ func Test_Common_handleMsgRegisterAccount(t *testing.T) {
 		},
 		"fail domain name does not exist": {
 			BeforeTest: func(t *testing.T, k keeper.Keeper, ctx sdk.Context, mocks *keeper.Mocks) {
-				// set regexp match nothing in blockchain targets
+				// set regexp match nothing in blockchain resources
```
Comment (davepuchyr) [inline x/domain/account_handlers_test.go:1315]: // set regexp match nothing in resources

File: x/domain/account_handlers_test.go [hunk 8]
```diff
@@ -1354,15 +1345,15 @@ func Test_Common_handleMsgRegisterAccount(t *testing.T) {
 		"fail only owner of domain with superuser can register accounts": {
 			TestBlockTime: 1,
 			BeforeTest: func(t *testing.T, k keeper.Keeper, ctx sdk.Context, mocks *keeper.Mocks) {
-				// set regexp match nothing in blockchain targets
+				// set regexp match nothing in blockchain resources
```
Comment (davepuchyr) [inline x/domain/account_handlers_test.go:1348]: // set regexp match nothing in resources

File: x/domain/client/cli/tx.go [hunk 9]
```diff
@@ -183,10 +183,10 @@ func getCmdTransferAccount(cdc *codec.Codec) *cobra.Command {
 	return cmd
 }
 
-func getCmdReplaceAccountTargets(cdc *codec.Codec) *cobra.Command {
+func getmCmdReplaceAccountResources(cdc *codec.Codec) *cobra.Command {
```
Comment (orkunkl) [inline x/domain/client/cli/tx.go:186]: Rename this to getCmdReplaceAccountResources

File: x/domain/controllers/account/account.go [hunk 10]
```diff
@@ -325,34 +325,34 @@ func (a *Account) deletableBy(addr sdk.AccAddress) error {
...
-	validBlockchainID := regexp.MustCompile(a.conf.ValidBlockchainID)
-	validBlockchainAddress := regexp.MustCompile(a.conf.ValidBlockchainAddress)
-	// create blockchain targets set to identify duplicates
-	sets := make(map[string]struct{}, len(targets))
-	// iterate over targets to check their validity
-	for _, target := range targets {
-		// check if blockchain ID was already specified
-		if _, ok := sets[target.ID]; ok {
-			return sdkerrors.Wrapf(types.ErrInvalidBlockchainTarget, "duplicate blockchain ID %s", target.ID)
+	validBlockchainID := regexp.MustCompile(a.conf.ValidResourceURI)
```
Comment (davepuchyr) [inline x/domain/controllers/account/account.go:338]: `validBlockchainID` should be `validURI`

File: x/domain/controllers/account/account.go [hunk 11]
```diff
@@ -325,34 +325,34 @@ func (a *Account) deletableBy(addr sdk.AccAddress) error {
...
+	validBlockchainAddress := regexp.MustCompile(a.conf.ValidResourceContent)
```
Comment (davepuchyr) [inline x/domain/controllers/account/account.go:339]: `validBlockchainAddress` should be `validResource`

File: x/domain/controllers/account/account.go [hunk 12]
```diff
@@ -432,19 +432,19 @@ func (a *Account) gracePeriodFinished() error {
...
-func BlockchainTargetLimitNotExceeded(targets []types.BlockchainAddress) ControllerFunc {
+func ResourceLimitNotExceeded(resources []types.Resource) ControllerFunc {
 	return func(ctrl *Account) error {
-		return ctrl.blockchainTargetLimitNotExceeded(targets)
+		return ctrl.blockchainResourceLimitNotExceeded(resources)
 	}
 }
 
-func (a *Account) blockchainTargetLimitNotExceeded(targets []types.BlockchainAddress) error {
+func (a *Account) blockchainResourceLimitNotExceeded(resources []types.Resource) error {
```
Comment (davepuchyr) [inline x/domain/controllers/account/account.go:441]: `blockchainResourceLimitNotExceeded` should be `resourceLimitNotExceeded`

File: x/domain/keeper/index.go [hunk 13]
```diff
@@ -13,8 +13,8 @@ var ownerToAccountPrefix = []byte{0x04}
 // ownerToDomainPrefix is the prefix that matches owners to domains
 var ownerToDomainPrefix = []byte{0x05}
 
-// blockchainTargetsPrefix is the prefix used to index targets to account
-var blockchainTargetsPrefix = []byte{0x06}
+// blockchainResourcesPrefix is the prefix used to index resources to account
+var blockchainResourcesPrefix = []byte{0x06}
```
Comment (davepuchyr) [inline x/domain/keeper/index.go:17]: `blockchainResourcesPrefix` should be `resourcesPrefix`

File: x/domain/keeper/index.go [hunk 14]
```diff
@@ -39,20 +39,20 @@ func ownerToAccountIndexStore(kvstore sdk.KVStore, addr sdk.AccAddress) (index.S
 	return index.NewAddressIndex(indexPrefixedStore, ownerToAccountPrefix, addr)
 }
 
-// blockchainTargetIndexedStore returns the store used to index blockchain targets
-func blockchainTargetIndexedStore(store sdk.KVStore, target types.BlockchainAddress) (index.Store, error) {
+// blockchainResourcesIndexStore returns the store used to index accounts resources
+func blockchainResourcesIndexStore(store sdk.KVStore, resource types.Resource) (index.Store, error) {
```
Comment (davepuchyr) [inline x/domain/keeper/index.go:43]: `blockchainResourcesIndexStore` should be `resourcesIndexStore`

File: x/domain/controllers/account/account.go (hunk 10 above)
Comment (davepuchyr) [reply x/domain/controllers/account/account.go]: @fdymylja , `ValidResourceURI` should be `ValidURI`, otherwise - there's redundancy on "resource".

**Pull request merged**